packages/project-themes/lektor_project_themes.py

```diff
@@ -0,0 +1,307 @@
+# -*- coding: utf-8 -*-
+
+"""Plugin to add Themes page"""
+
+import base64
+import os
+import traceback
+from configparser import ConfigParser
+from shutil import copyfile
+from typing import List
+
+import requests
+
+from lektor.pluginsystem import Plugin
+
+GITHUB_THEMES_REPO = 'lektor/lektor-themes'
+
+
+class ProjectThemesPlugin(Plugin):
+    """Plugin that generates Themes pages"""
+
+    name = 'Project Themes'
+    description = 'Build Themes page'
+
+    def on_setup_env(self, **extra):
+        self._create_themes_content()
+
+    def _create_themes_content(self):
+        """Creates content for the themes page"""
+        path_content = os.path.join(self.env.root_path, 'content')
+        path_themes = os.path.join(path_content, 'themes')
+
+        if self._skip_creating_all_content(path_themes):  # Skip redownloading themes for dev speed
+            return
+
+        os.makedirs(path_themes, exist_ok=True)
+
+        self._add_contents_lr(path_themes)
+        self._add_header(path_content, path_themes)
+        self._add_themes(path_themes)
+
+    @staticmethod
+    def _skip_creating_all_content(path_themes):
+        """
+        Determine if the contents for themes page already exists
+
+        :param str path_themes: Path to themes folder
+        :return: Plugin should skip creating anything
+        :rtype: bool
+        """
+        return os.path.isdir(path_themes)
+
+    @staticmethod
+    def _add_contents_lr(path_themes):
+        """
+        Adds contents.lr file
+
+        :param str path_themes: Path to themes folder
+        """
+        path = os.path.join(path_themes, 'contents.lr')
+        with open(path, 'w', encoding='utf-8') as f:
+            f.write('_model: themes')
+
+    @staticmethod
+    def _add_header(path_content, path_themes):
+        """
+        Adds header image to themes pages
+
+        :param str path_content: Path to content folder
+        :param str path_themes: Path to themes folder
+        """
+        path_header_src = os.path.join(path_content, 'header.jpg')
+        path_header_dest = os.path.join(path_themes, 'header.jpg')
+        copyfile(path_header_src, path_header_dest)
+
+    def _add_themes(self, path_themes):
+        """
+        Add each theme
+
+        :param str path_themes: Path to themes folder
+        """
+        api_themes = self._get_themes_via_api()
+
+        for t in api_themes:
+            name = t['name']
+            git_url = t['git_url']
+            try:
+                theme = _Theme(name, git_url)
+            except Exception:  # No exception from a theme should prevent website from building
+                print("FAILED TO ADD THEME")
```

Comment:
Let's make this stand out a bit more. :) It would be way easier to pick out styling at least this line. https://github.com/lektor/lektor/blob/master/lektor/cli.py#L345